src/Bicep.Core/Diagnostics/DiagnosticBuilder.cs

```diff
@@ -1232,6 +1232,86 @@ public ErrorDiagnostic UnknownModuleReferenceScheme(string badScheme, ImmutableA
+
+            public ErrorDiagnostic InvalidTemplateSpecAliasResourceGroupNullOrUndefined(string aliasName, string configurationResourceName) => new(
+                TextSpan,
+                "BCP215",
+                $"The Template Spec module alias \"{aliasName}\" in the Bicep configuration \"{configurationResourceName}\" is in valid. The \"resourceGroup\" property cannot be null or undefined.");
+
+            public ErrorDiagnostic InvalidOciArtifactModuleAliasRegistryNullOrUndefined(string aliasName, string configurationResourceName) => new(
+                TextSpan,
+                "BCP216",
+                $"The OCI artifact alias \"{aliasName}\" in the Bicep configuration \"{configurationResourceName}\" is invalid. The \"registry\" property cannot be null or undefined.");
+
+            public ErrorDiagnostic InvalidTemplateSpecReferenceInvalidSubscirptionId(string subscriptionId, string referenceValue) => new(
+                TextSpan,
+                "BCP217",
+                $"The specified Template Spec reference \"{referenceValue}\" is invalid. The subscription ID \"{subscriptionId}\" in is not a GUID."); 
+
+            public ErrorDiagnostic InvalidTemplateSpecReferenceResourceGroupNameTooLong(string resourceGroupName, string referenceValue, int maximumLength) => new(
+                TextSpan,
+                "BCP218",
+                $"The specified Template Spec reference \"{referenceValue}\" is invalid. The resource group name \"{resourceGroupName}\" exceeds the maximum length of {maximumLength} characters."); 
+
+            public ErrorDiagnostic InvalidTemplateSpecReferenceInvalidResourceGroupName(string resourceGroupName, string referenceValue) => new(
+                TextSpan,
+                "BCP219",
+                $"The specified Template Spec reference \"{referenceValue}\" is invalid .The resource group name \"{resourceGroupName}\" is invalid. Valid characters are alphanumeric, \".\", \"_\", \"-\", \"(\", or \")\", but the resource group name cannot end with \".\"."); 
+
+            public ErrorDiagnostic InvalidTemplateSpecReferenceTemplateSpecNameTooLong(string templateSpecName, string referenceValue, int maximumLength) => new(
+                TextSpan,
+                "BCP220",
+                $"The specified Template Spec reference \"{referenceValue}\" is invalid. The Template Spec name \"{templateSpecName}\" exceeds the maximum length of {maximumLength} characters."); 
+
+            public ErrorDiagnostic InvalidTemplateSpecReferenceInvalidTemplateSpecName(string templateSpecName, string referenceValue) => new(
+                TextSpan,
+                "BCP221",
+                $"The specified Template Spec reference \"{referenceValue}\" is invalid .The Template Spec name \"{templateSpecName}\" is invalid. Valid characters are alphanumeric, \".\", \"_\", \"-\", \"(\", or \")\", but the Template Spec name cannot end with \".\"."); 
```

Comment:
>alphanumeric [](http://example.com/codeflow?start=168&length=12)

Do template spec names and version actually have this restriction?

Reply:
You're using `\w` in the regex for parsing these things, which includes some Unicode classes as well: [https://docs.microsoft.com/en-us/dotnet/standard/base-types/character-classes-in-regular-expressions#word-character-w](https://docs.microsoft.com/en-us/dotnet/standard/base-types/character-classes-in-regular-expressions#word-character-w)

Reply:
Synced offline - confirmed via tests that this matches the service behavior so will keep it as is.